File: TestCase_nArmedBandit/env_nArmedBandit.m
function [s,r] = env_nArmedBandit(a)
%ENV_NARMEDBANDIT environment to respond to an (a)ction with a new (s)tate and a
%(r)eward

global env;

% For the n-armed bandit example, the state does not vary with time
s = 0;

prob_win = env.prob(a);
if rand < prob_win
    r = env.reward(1);
else
    r = env.reward(2);
end

end

File: TestCase_nArmedBandit/env_nArmedBandit.py
import numpy as np
import random


class Env(object):
    def __init__(self, n, randseed):
        random.seed(randseed)
        self.prob = np.array([random.uniform(0, 1) for _ in range(n)])
        self.reward = np.array([1, 0])

    def payout(self, action):
        prob_win = self.prob[action]
        if random.uniform(0, 1) < prob_win:
            r = self.reward[0]
        else:
            r = self.reward[1]
        return r
